### DISCUSSION TITLE
Attributes in class definition

### DISCUSSION THREAD

**Discussion by antonioturdo** [CuyZ/Valinor#525] [General]
HI. I just wanted to share a problem in using the library, possibly as a starting point for future developments.

My use case involves a very large DTO (50+ fields) to which I need to map the query string parameters of an http request.

Each field of the DTO has the OpenAPI attribute, with the consequence that the cached definition of the class is really large (in my case more than 50K rows and 2MB in size). From tests carried out, the presence of the attributes determines a significant deterioration in performance.

I know I could avoid using those attributes in prod, but for reasons complicated to explain here, it's not as easy as it seems.

In any case, even though I read your position [here](https://github.com/CuyZ/Valinor/issues/393), I think it would be useful to be able to have more control over what ends up in the class definition, also because the same problem could arise with other attributes that are impossible to move to `require-dev`.

**Comment by romm** [comment]
Hi @antonioturdo thanks for this report. Could you take a look at #530, try it on your side and tell me if the generated cache entries are lighter?

**Comment by antonioturdo** [comment]
Hi @romm yes, I can confirm that the generated cache is lighter, and that it no longer contains the OpenAPI attributes.

Thank you for taking my report into consideration, and for implementing significant changes in a short time.

**Reply by romm** [reply]
Glad to hear that!
